xl2times/transforms.py

```diff
@@ -2703,9 +2722,29 @@ def apply_transform_tables(
                 new_rows["submodule"] = row["submodule"]
                 new_tables.append(new_rows)
 
-            # Add new rows to table
-            new_tables.append(tables[Tag.fi_t])
-            tables[Tag.fi_t] = pd.concat(new_tables, ignore_index=True)
+            generated_records.append(pd.concat(new_tables, ignore_index=True))
+
+        if generated_records:
+            module_data = pd.concat(generated_records, ignore_index=True)
+            module_type = module_data["module_type"].iloc[0]
+            # Explode process and commodity columns and remove invalid rows
+            for obj in ["process", "commodity"]:
+                if obj in module_data.columns:
+                    module_data = module_data.explode(obj, ignore_index=True)
+                    if module_type in {"base", "subres"}:
+                        valid_objs = (
+                            obj_by_module[obj]
+                            .get(data_module, set())
+                            .union(obj_suppl[obj])
+                        )
+                        drop = ~module_data[obj].isin(valid_objs) & module_data[
+                            "attribute"
+                        ].isin(attr_with_obj[obj])
+                        if any(drop):
+                            module_data = module_data[~drop]
```

Comment:
It might work without the if-condition too?

Reply:
Thanks! It does. 😄
I've added an if-condition to check whether module_data is empty before concatenating though.